src/main/java/gov/nist/oar/rmm/repositories/impl/VersionReleasesetsRepositoryImpl.java

```diff
@@ -0,0 +1,157 @@
+package gov.nist.oar.rmm.repositories.impl;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.util.MultiValueMap;
+
+import com.mongodb.client.AggregateIterable;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.model.Filters;
+
+import gov.nist.oar.rmm.config.AppConfig;
+import gov.nist.oar.rmm.config.MongoConfig;
+import gov.nist.oar.rmm.exceptions.ResourceNotFoundException;
+import gov.nist.oar.rmm.repositories.VersionReleasesetsRepository;
+import gov.nist.oar.rmm.utilities.ProcessRequest;
+
+@Service
+public class VersionReleasesetsRepositoryImpl implements VersionReleasesetsRepository {
+
+    private Logger logger = LoggerFactory.getLogger(VersionReleasesetsRepositoryImpl.class);
+
+    @Autowired
+    MongoConfig mconfig;
+
+    @Autowired
+    AppConfig appconfig;
+
+    /**
+     * Search Versions dataset to get version of the record This can be queried
+     * using key,value pairs as parameters
+     * 
+     * @param param key,value parameters
+     * @param p     pagination
+     * @return
+     */
+    @Override
+    public Document findVersion(MultiValueMap<String, String> param, Pageable p) {
+
+        return this.queryResults(param, p, "Versions");
```

Comment:
Use `mconfig.versions` here instead of hard-coded name; see comments below.

Reply:
My config class does not return the string of collection name instead it returns the collection object only. I will check whether I can extract only name from it. This code works for me fine. Can you check what names you entered in database for these collections.